### PR TITLE
docs: translation for forwardRef

### LINKED ISSUE #470: `forwardRef`
Page: https://id.react.dev/reference/react/forwardRef

Assignee: @r17x

### PR DESCRIPTION
**Pull request by r17x**
Translation for forwardRef reference

Closes #470

### CODE REVIEW

File: src/content/reference/react/forwardRef.md [hunk 1]
```diff
@@ -30,26 +30,26 @@ const MyInput = forwardRef(function MyInput(props, ref) {
...
+`forwardRef` mengembalikan komponen React yang dapat Anda *render* di JSX. Tidak seperti komponen React yang didefinisikan sebagai fungsi biasa, komponen yang dikembalikan oleh `forwardRef` juga dapat menerima *prop* `ref`.
 
-#### Caveats {/*caveats*/}
+#### Peringatan {/*caveats*/}
 
-* In Strict Mode, React will **call your render function twice** in order to [help you find accidental impurities.](#my-initializer-or-updater-function-runs-twice) This is development-only behavior and does not affect production. If your render function is pure (as it should be), this should not affect the logic of your component. The result from one of the calls will be ignored.
+* Dalam Mode Ketat, React akan **memanggil fungsi* render Anda dua kali** untuk [membantu Anda menemukan ketidakmurnian yang tidak disengaja.] (#my-initializer-or-updater-function-runs-two times) Ini adalah perilaku khusus pengembangan dan tidak mempengaruhi produksi. Jika fungsi render Anda murni (sebagaimana mestinya), hal ini tidak akan mempengaruhi logika komponen Anda. Hasil dari salah satu pemanggilan akan diabaikan.
```
Comment (agprst) [inline src/content/reference/react/forwardRef.md:45]:
```suggestion
* Dalam Mode Ketat, React akan **memanggil fungsi *render* Anda dua kali** untuk [membantu Anda menemukan ketidakmurnian yang tidak disengaja.] (#my-initializer-or-updater-function-runs-twice) Ini adalah perilaku khusus pengembangan dan tidak mempengaruhi produksi. Jika fungsi render Anda murni (sebagaimana mestinya), hal ini tidak akan mempengaruhi logika komponen Anda. Hasil dari salah satu pemanggilan akan diabaikan.
```

File: src/content/reference/react/forwardRef.md [hunk 2]
```diff
@@ -64,21 +64,21 @@ const MyInput = forwardRef(function MyInput(props, ref) {
 
 #### Parameters {/*render-parameters*/}
 
-* `props`: The props passed by the parent component.
+* `props`: *props* yang dioperkan oleh komponen induk.
 
-* `ref`:  The `ref` attribute passed by the parent component. The `ref` can be an object or a function. If the parent component has not passed a ref, it will be `null`. You should either pass the `ref` you receive to another component, or pass it to [`useImperativeHandle`.](/reference/react/useImperativeHandle)
+* `ref`: Atribut `ref` yang dioper oleh komponen induk. `ref` dapat berupa objek atau fungsi. Jika komponen induk tidak mengoper *ref*, maka akan menjadi `null`. Anda harus mengoper `ref` yang Anda terima ke komponen lain, atau mengopernya ke [`useImperativeHandle`.] (/reference/react/useImperativeHandle)
```
Comment (agprst) [inline src/content/reference/react/forwardRef.md:69]:
```suggestion
* `ref`: Atribut `ref` yang dioper oleh komponen induk. `ref` dapat berupa objek atau fungsi. Jika komponen induk tidak mengoper *ref*, maka akan menjadi `null`. Anda harus mengoper `ref` yang Anda terima ke komponen lain, atau mengopernya ke [`useImperativeHandle`.](/reference/react/useImperativeHandle)
```

File: src/content/reference/react/forwardRef.md [hunk 3]
```diff
@@ -463,25 +464,25 @@ input {
 
 </Sandpack>
 
-[Read more about using imperative handles.](/reference/react/useImperativeHandle)
+[Baca lebih lanjut tentang menggunakan pegangan imperatif.] (/reference/react/useImperativeHandle)
```
Comment (agprst) [inline src/content/reference/react/forwardRef.md:467]:
```suggestion
[Baca lebih lanjut tentang menggunakan pegangan imperatif.](/reference/react/useImperativeHandle)
```

**Comment by agprst** [comment]
added suggestions for some broken links

File: src/content/reference/react/forwardRef.md [hunk 4]
```diff
@@ -30,26 +30,26 @@ const MyInput = forwardRef(function MyInput(props, ref) {
...
+`forwardRef` mengembalikan komponen React yang dapat Anda *render* di JSX. Tidak seperti komponen React yang didefinisikan sebagai fungsi biasa, komponen yang dikembalikan oleh `forwardRef` juga dapat menerima *prop* `ref`.
 
-#### Caveats {/*caveats*/}
+#### Peringatan {/*caveats*/}
 
-* In Strict Mode, React will **call your render function twice** in order to [help you find accidental impurities.](#my-initializer-or-updater-function-runs-twice) This is development-only behavior and does not affect production. If your render function is pure (as it should be), this should not affect the logic of your component. The result from one of the calls will be ignored.
+* Dalam Mode Ketat, React akan **memanggil fungsi* render Anda dua kali** untuk [membantu Anda menemukan ketidakmurnian yang tidak disengaja.] (#my-initializer-or-updater-function-runs-twice) Ini adalah perilaku khusus pengembangan dan tidak mempengaruhi produksi. Jika fungsi render Anda murni (sebagaimana mestinya), hal ini tidak akan mempengaruhi logika komponen Anda. Hasil dari salah satu pemanggilan akan diabaikan.
```
Comment (agprst) [inline src/content/reference/react/forwardRef.md:45]:
```suggestion
* Dalam Mode Ketat, React akan **memanggil fungsi *render* Anda dua kali** untuk [membantu Anda menemukan ketidakmurnian yang tidak disengaja.] (#my-initializer-or-updater-function-runs-twice) Ini adalah perilaku khusus pengembangan dan tidak mempengaruhi produksi. Jika fungsi render Anda murni (sebagaimana mestinya), hal ini tidak akan mempengaruhi logika komponen Anda. Hasil dari salah satu pemanggilan akan diabaikan.
```
*italicize* `render`

**Review by mhaidarhanif — APPROVED** [review]
It's good. @resir014 can confirm.

File: src/content/reference/react/forwardRef.md [hunk 5]
```diff
@@ -131,15 +131,16 @@ function Form() {
...
+Komponen `MyInput` meneruskan *ref* tersebut ke tag peramban `<input>`. Hasilnya, komponen `Form` dapat mengakses *DOM node* `<input>` tersebut dan memanggil fungsi [`focus()`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/focus) di atasnya.
+
+Perlu diingat bahwa mengekspos *ref* ke *DOM node* di dalam komponen Anda akan mempersulit untuk mengubah internal komponen Anda di kemudian hari. Anda biasanya akan mengekspos *DOM node* dari komponen tingkat rendah yang dapat digunakan kembali seperti tombol atau input teks, tetapi Anda tidak akan melakukannya untuk komponen tingkat aplikasi seperti avatar atau komentar.
 
-Keep in mind that exposing a ref to the DOM node inside your component makes it harder to change your component's internals later. You will typically expose DOM nodes from reusable low-level components like buttons or text inputs, but you won't do it for application-level components like an avatar or a comment.
 
 <Recipes title="Examples of forwarding a ref">
 
-#### Focusing a text input {/*focusing-a-text-input*/}
+#### memfokuskan input teks {/*focusing-a-text-input*/}
```
Comment (mazipan) [inline src/content/reference/react/forwardRef.md:141]:
```suggestion
#### Memfokuskan input teks {/*focusing-a-text-input*/}
```

File: src/content/reference/react/forwardRef.md [hunk 6]
```diff
@@ -131,15 +131,16 @@ function Form() {
...
 
-Clicking the button will focus the input. The `Form` component defines a ref and passes it to the `MyInput` component. The `MyInput` component forwards that ref to the browser `<input>`. This lets the `Form` component focus the `<input>`.
+Mengklik tombol akan memfokuskan input. Komponen `Form` mendefinisikan sebuah ref dan meneruskannya ke komponen `MyInput`. Komponen `MyInput` meneruskan *ref* tersebut ke tag peramban `<input>`. Hal ini memungkinkan komponen `Form` memfokuskan `<input>`.
```
Comment (mazipan) [inline src/content/reference/react/forwardRef.md:143]:
```suggestion
Mengeklik tombol akan memfokuskan input. Komponen `Form` mendefinisikan sebuah ref dan meneruskannya ke komponen `MyInput`. Komponen `MyInput` meneruskan *ref* tersebut ke tag peramban `<input>`. Hal ini memungkinkan komponen `Form` memfokuskan `<input>`.
```

File: src/content/reference/react/forwardRef.md [hunk 7]
```diff
@@ -191,9 +192,9 @@ input {
 
 <Solution />
 
-#### Playing and pausing a video {/*playing-and-pausing-a-video*/}
+#### Memutar dan menjeda video {/*playing-and-pausing-a-video*/}
 
-Clicking the button will call [`play()`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/play) and [`pause()`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/pause) on a `<video>` DOM node. The `App` component defines a ref and passes it to the `MyVideoPlayer` component. The `MyVideoPlayer` component forwards that ref to the browser `<video>` node. This lets the `App` component play and pause the `<video>`.
+Mengklik tombol akan memanggil [`play()`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/play) dan [`pause()`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/pause) pada *DOM node* `<video>`. Komponen `Aplikasi` mendefinisikan sebuah ref dan meneruskannya ke komponen `MyVideoPlayer`. Komponen `MyVideoPlayer` meneruskan *ref* tersebut ke *node* `<video>` pada peramban. Hal ini memungkinkan komponen `Aplikasi` memainkan dan menjeda `<video>`.
```
Comment (mazipan) [inline src/content/reference/react/forwardRef.md:197]:
```suggestion
Mengeklik tombol akan memanggil [`play()`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/play) dan [`pause()`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/pause) pada *DOM node* `<video>`. Komponen `Aplikasi` mendefinisikan sebuah ref dan meneruskannya ke komponen `MyVideoPlayer`. Komponen `MyVideoPlayer` meneruskan *ref* tersebut ke *node* `<video>` pada peramban. Hal ini memungkinkan komponen `Aplikasi` memainkan dan menjeda `<video>`.
```

File: src/content/reference/react/forwardRef.md [hunk 8]
```diff
@@ -381,7 +382,7 @@ const MyInput = forwardRef(function MyInput(props, ref) {
 });
 ```
 
-Pass the `ref` you received to [`useImperativeHandle`](/reference/react/useImperativeHandle) and specify the value you want to expose to the `ref`:
+Berikan `ref` yang Anda terima ke [`useImperativeHandle`](/reference/react/useImperativeHandle) dan tentukan nilai yang ingin Anda pajankan ke `ref`:
```
Comment (mazipan) [inline src/content/reference/react/forwardRef.md:385]: Not sure soal ini, tapi sepertinya:

```suggestion
Berikan `ref` yang Anda terima ke [`useImperativeHandle`](/reference/react/useImperativeHandle) dan tentukan nilai yang ingin Anda ekspos ke `ref`:
```

File: src/content/reference/react/forwardRef.md [hunk 9]
```diff
@@ -463,25 +464,25 @@ input {
 
 </Sandpack>
 
-[Read more about using imperative handles.](/reference/react/useImperativeHandle)
+[Baca lebih lanjut tentang menggunakan pegangan imperatif.](/reference/react/useImperativeHandle)
```
Comment (mazipan) [inline src/content/reference/react/forwardRef.md:467]: Pakai italic aja apa?

```suggestion
[Baca lebih lanjut tentang menggunakan *imperative handle*.](/reference/react/useImperativeHandle)
```

File: src/content/reference/react/forwardRef.md [hunk 10]
```diff
@@ -463,25 +464,25 @@ input {
...
+**Jangan terlalu sering menggunakan refs.** Anda hanya boleh menggunakan *refs* untuk perilaku *imperatif* yang tidak dapat Anda ungkapkan sebagai *props*: misalnya, menggulir ke sebuah *node*, memfokuskan sebuah *node*, memicu sebuah animasi, memilih teks, dan sebagainya.
 
-**If you can express something as a prop, you should not use a ref.** For example, instead of exposing an imperative handle like `{ open, close }` from a `Modal` component, it is better to take `isOpen` as a prop like `<Modal isOpen={isOpen} />`. [Effects](/learn/synchronizing-with-effects) can help you expose imperative behaviors via props.
+**Jika Anda dapat mengekspresikan sesuatu sebagai *prop*, Anda tidak boleh menggunakan *ref*.** Sebagai contoh, daripada mengekspos penanganan imperatif seperti `{ buka, tutup }` dari komponen `Modal`, lebih baik menggunakan `isOpen` sebagai *prop* seperti `<Modal isOpen = {isOpen} />`. [Effects](/learn/synchronizing-with-effects) dapat membantu Anda mengekspos perilaku imperatif melalui  *props*.
```
Comment (mazipan) [inline src/content/reference/react/forwardRef.md:473]: Not sure apakah kita perlu menterjemahkan `open, close` di sini

Comment (r17x) [reply src/content/reference/react/forwardRef.md:473]: Sorry for my bad. Udah balik lagi

File: src/content/reference/react/forwardRef.md (hunk 9 above)
Comment (r17x) [reply src/content/reference/react/forwardRef.md:467]: Ku samakan dengan translation yang lain dimana `handle imperative` menjadi `penanganan imperatif`

**Pull request merged**